File: framework/java/src/main/java/soda/unittest/__Bootstrap__.java
package soda.unittest;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.work.TestWork;

import static soda.unittest.LoggerHelper.logger;

class Solution {}

public class __Bootstrap__ implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "METHOD");
        // work.setValidator((e, r) -> {...});
        work.setCompareSerial(true);
        // work.setArgumentParser(index, a -> { ... });
        // work.setResultParser(r -> { ... });
        // work.setResultSerializer(r -> {...});
        return work;
    }

    public static void main(String[] args) throws Exception {
        new __Bootstrap__().get().run();
    }

}

File: framework/scala/src/main/scala/soda/unittest/__Bootstrap__.scala
package soda.unittest

import java.util._
import java.util.stream._

import soda.leetcode._

import soda.unittest.LoggerHelper.logger

// setp [1]: implement object Solution
// object Solution

// step [2]: implement test job
class __Bootstrap__ extends JobTemplate[Object, Object] {

    override def execute(req: TestRequest, resp: TestResponse): Object = {
        // TODO
        throw new RuntimeException("Not implemented");
    }

    override def serialize(res: Object): Object = {
        res
    }
    
    override def validate(req: TestRequest, resp: TestResponse): Boolean = {
        req.getExpected(classOf[Object]).equals(resp.getResult(classOf[Object]))
    }

}

object __Bootstrap__ extends App {
    new ConsoleRunner().run(new __Bootstrap__())
}
